Source/PluginEditor.cpp

```diff
@@ -158,4 +159,64 @@ namespace reverb
         }
 	}
 
+    /**
+    * @brief   Callback function to open the popup menu asynchronously.
+    *
+    * @details This function will open a popup menu when the user clicks
+    *          on the IR selection's button. The first item in the menu
+    *          will let the user choose a custom IR from a file browser,
+    *          while the other items will display the IRs from the IR bank.
+    * @param   [in] clickedButton  Pointer to the button clicked by the user.
+    */
+    void AudioProcessorEditor::buttonClicked(juce::Button* clickedButton)
+    {
+        if (clickedButton != nullptr && clickedButton == &headerBlock.irChoice) {
+            juce::PopupMenu m;
+            m.addItem(1, "Load custom IR...");
+            m.addSeparator();
+            for (auto irFile : IRBank::getInstance().buffers) {
+                m.addItem(m.getNumItems() + 1, irFile.first);
+            }
+
+            m.showMenuAsync(juce::PopupMenu::Options(),
+                juce::ModalCallbackFunction::forComponent(menuCallback, &headerBlock));
+        }
+    }
+
+    /**
+    * @brief   Callback function to open the popup menu asynchronously. 
+    *
+    * @details This function will open a file browser if the user clicked on the first
+    *          item in the dropdown menu. Any other item will load the appropriate IR
+    *          from the IR bank.
+    * @param   [in] result        The index of the item selected by the user.
+    *          [in] headerBlock   Pointer to the header block.
+    */
+    void AudioProcessorEditor::menuCallback(int result, UIHeaderBlock* headerBlock)
+    {
+        if (headerBlock != nullptr && result != 0) {
+            if (result == 1) {
+                juce::AudioFormatManager formatManager;
+                formatManager.registerBasicFormats();
+                // Accepts the following file types: .wav, .bwf, .aiff, .flac, .ogg, .mp3, .wmv, .asf, .wm, .wma
+                juce::FileChooser fileChooser("Select an impulse response",
+                    juce::File::getCurrentWorkingDirectory(),
+                    formatManager.getWildcardForAllFormats(),
+                    false);
+                if (fileChooser.browseForFileToOpen()) {
+                    juce::File selectedFile = fileChooser.getResult();
+                    headerBlock->parameters.state.getChildWithName(AudioProcessor::PID_IR_FILE_CHOICE)
+                        .setProperty("value", selectedFile.getFullPathName(), nullptr);
+                    headerBlock->irChoice.setButtonText(selectedFile.getFullPathName());
+                }
+            }
+            else {
+                juce::String irFile = std::next(IRBank::getInstance().buffers.begin(), result - 2)->first;
+                headerBlock->parameters.state.getChildWithName(AudioProcessor::PID_IR_FILE_CHOICE)
+                    .setProperty("value", irFile, nullptr);
+                headerBlock->irChoice.setButtonText(irFile);

```

Comment:
headerBlock-> parameters & headerBlock-> irChoice are common in both this **else** and the previous **if** with the difference being how selectedFile & irFile are defined. Should be a simple fix.